### PR TITLE
fix(toggle-button): add missing ARIA and data props (FE-4446)

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour
Add data props to ButtonToggle.
Add ARIA props to ButtonToggle.
Change the role of ButtonToggleGroup to `radiogroup`.
Add `radio` role to ButtonToggle.

### Current behaviour
Data props could not be passed.
No ARIA props on ButtonToggle.

### Checklist

- [x] Commits follow our style guide
~~- [ ] Related issues linked in commit messages if required~~
~~- [ ] Screenshots are included in the PR if useful~~
~~- [ ] All themes are supported if required~~
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
~~- [ ] Storybook added or updated if required~~
~~- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required~~
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/button-toggle/button-toggle.component.js [hunk 1]
```diff
@@ -82,8 +95,18 @@ const ButtonToggle = (props) => {
 };
 
 ButtonToggle.propTypes = {
+  /** Prop to specify the aria-label of the component */
+  "aria-label": PropTypes.string,
+  /** Prop to specify the aria-labeledby property of the component */
```
Comment (mkrds) [inline src/components/button-toggle/button-toggle.component.js:100]:
```suggestion
  /** Prop to specify the aria-labelledby property of the component */
```

File: src/components/button-toggle/button-toggle.component.js [hunk 2]
```diff
@@ -48,11 +54,18 @@ const ButtonToggle = (props) => {
 
   return (
     <StyledButtonToggle
-      data-component="button-toggle"
+      aria-checked={checked}
```
Comment (mkrds) [inline src/components/button-toggle/button-toggle.component.js:57]: Shouldn't `aria-checked` be applied to the `ButtonToggleInput` ?

**Review by DipperTheDan — COMMENTED** [review]
Looks like these changes have introduced a new accessibility violation. This is in the `default` example.

Screenshot 2021-12-01 at 09 48 19

File: src/components/button-toggle/button-toggle.component.js (hunk 2 above)
Comment (grabkowski) [reply src/components/button-toggle/button-toggle.component.js:57]: the input is hidden, so it should be ignored by readers

Comment (grabkowski) [reply src/components/button-toggle/button-toggle.component.js]: looks like the VoiceOver utility is reading that input anyway, so this aria prop is redundant

**Review by mkrds — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Pull request merged**